### DISCUSSION TITLE
Customize the table of contents of a doc

### DISCUSSION THREAD

**Discussion by adam8451** [facebook/docusaurus#7775] [Q&A] [answered]
Hi everyone,

In a given doc, is there an easy way to customize the table of contents shown at the right of the page?

**Use case**. Unclutter the left hand sidebar in the following situation:
- Folders have a dense triply-nested structure (tedious navigation process)
- Each doc is succinct (absence of "natural" ToC)

**Benefit**. Transition from:
- Crowded left sidebar
```
folder_1
  subfolder_1
    subsubfolder_1
      item_1
      ...
      item_n
    ...
    subsubfolder_n
      item_1
      ...
      item_n    
  ...
  subfolder_n
    subsubfolder_1
      item_1
      ...
      item_n
    ...
    subsubfolder_n
      item_1
      ...
      item_n
...
folder_n
  subfolder_1
    subsubfolder_1
      item_1
      ...
      item_n
    ...
    subsubfolder_n
      item_1
      ...
      item_n    
  ...
  subfolder_n
    subsubfolder_1
      item_1
      ...
      item_n
    ...
    subsubfolder_n
      item_1
      ...
      item_n
```
- Empty right sidebar
```
```

to:
- Reasonably-populated left sidebar
```
folder_1
  subfolder_1   
  ...
  subfolder_n
...
folder_n
  subfolder_1 
  ...
  subfolder_n
```

- Reasonably-populated right sidebar
```
subsubfolder_1
  item_1
  ...
  item_n
...
subsubfolder_n
  item_1
  ...
  item_n
```

**Requirements**. Ideally, the custom "table of contents" of a given `subfolder_i` would have the following characteristics:
- It can be inferred from the folder structure
- Or be easily configurable, e.g. through an API that reads from a Markdown list of items

**Best try so far**. Tediously editing the built HTML.

Thanks in advance for your time and help!

**Comment by slorber** [comment, marked as answer]
- Each doc on the file-system has a dedicated page
- We navigate through the page headings with the TOC
- We navigate through pages using the sidebar
- The TOC gives control over heading levels, you can have 3 or 4 levels if you want through docs plugin options

We don't plan to implement features such as composing multiple file-system files into a single page. 
If you want to have that, you'd have to merge all your items into a single page upfront. 

Some workarounds if you really want to keep multiple file-system source files
- Have a pre-build script to create the final page with all items being a separate top-level heading (probably the easiest)
- Create a remark plugin to achieve what you want (may not be easy)
- Use MDX imports and partials to create the final page, but unfortunately the TOC does not work well with that yet and it's complex for us to fix it (see https://github.com/facebook/docusaurus/issues/3915)

**Reply by adam8451** [reply]
Thank you Sébastien for your thorough answer!
